crates/transparency/src/map/path.rs

```diff
@@ -28,15 +26,15 @@ impl Side {
     }
 }
 
-impl<D: SupportedDigest> Path<D> {
-    pub(crate) fn new<K: VisitBytes>(key: K) -> Self {
+impl<D: SupportedDigest, K: VisitBytes + Clone> Path<D, K> {
+    pub(crate) fn new(key: K) -> Self {
         let all = Hash::of(&key);
 
-        Self {
-            lhs: 0,
-            rhs: all.len() * 8,
-            all,
-        }
+        Self { key, index: 0, all }
+    }
+
+    pub fn key(&self) -> K {
```

Comment:
A key (i.e. ` LogId`) might have a clone that allocates, so can we return `&K` here so that the caller gets to decide if a clone is necessary?